src/back/glsl/features.rs

```diff
@@ -347,6 +355,25 @@ impl<'a, W> Writer<'a, W> {
             }
         }
 
+        let has_fma = self
+            .module
+            .functions
+            .iter()
+            .flat_map(|(_, f)| f.expressions.iter())
+            .chain(
+                self.module
+                    .entry_points
+                    .iter()
+                    .flat_map(|e| e.function.expressions.iter()),
+            )
+            .any(|(_, e)| match *e {
+                Expression::Math { fun, .. } if fun == MathFunction::Fma => true,
+                _ => false,
+            });
+        if has_fma {
+            self.features.request(Features::FMA);
```

Comment:
I don't think we want to do this. FMA can be trivially polyfilled. If it's not supported, the backend should still produce valid code, not error.

Reply:
Yep! Totally agree. I have a way to test this in `wgpu` so I can keep plugging away at it.